puller.Dockerfile

```diff
@@ -0,0 +1,21 @@
+# Build the inference-puller binary
```

Comment:
IMO, we should stop creating top level Dockerfiles. Perhaps we need something like
```
/cmd/manager/Dockerfile
/cmd/manager/main.go
/images/logger/Dockerfile
/images/logger/main.go
...
```

Reply:
Sure, lmk how you would prefer to structure it. I was mimicking the previous side-car implementations

Reply:
For go, let's try putting them with the main. Our python images are already this way. This way, each cmd/foo is effectively it's own package and can just have a Dockerfile.

Reply:
Sure I can move it over, on it